sda-commons-client-jersey-wiremock-testing/src/main/java/org/sdase/commons/client/jersey/wiremock/testing/WireMockExtension.java

```diff
@@ -0,0 +1,82 @@
+package org.sdase.commons.client.jersey.wiremock.testing;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.VerificationException;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.Options;
+import com.github.tomakehurst.wiremock.verification.LoggedRequest;
+import com.github.tomakehurst.wiremock.verification.NearMiss;
+import java.util.List;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Junit 5 replacement for {@link com.github.tomakehurst.wiremock.junit.WireMockRule} */
+public class WireMockExtension extends WireMockServer
+    implements BeforeEachCallback, AfterEachCallback {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WireMockExtension.class);
+
+  private final boolean failOnUnmatchedRequests;
+
+  public WireMockExtension(Options options) {
+    this(options, true);
+  }
+
+  public WireMockExtension(Options options, boolean failOnUnmatchedRequests) {
+    super(options);
+    this.failOnUnmatchedRequests = failOnUnmatchedRequests;
+  }
+
+  public WireMockExtension(int port) {
+    this(wireMockConfig().port(port));
+  }
+
+  public WireMockExtension(int port, Integer httpsPort) {
+    this(wireMockConfig().port(port).httpsPort(httpsPort));
+  }
+
+  public WireMockExtension() {
+    this(wireMockConfig());
```

Comment:
Can we use `dynamicPort()` as default here and mention that in JavaDoc?
I think the default constructor would cover most of the test cases then.